### PR TITLE
Disable checking inter pod affinity predicate if not necessary

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
Add disabling inter pod affinity predicate if only preferred / node affinity used.

### CODE REVIEW

**Comment by aleksandra-malinowska** [comment]
This has been tested manually for the following scenarios:
- deployment has pods using hard anti-affinity, verified that new nodes were added
- deployment has pods using soft anti-affinity, verified that predicate disabled log info is present, no new nodes are added as long as there's sufficient space on existing ones

File: cluster-autoscaler/core/utils.go [hunk 1]
```diff
@@ -412,23 +412,38 @@ func getPotentiallyUnneededNodes(context *AutoscalingContext, nodes []*apiv1.Nod
 	return result
 }
 
+func hasInterPodAffinity(affinity *apiv1.Affinity) bool {
```
Comment (mwielgus) [inline cluster-autoscaler/core/utils.go]: hasInterPod *HARD* Affinity

Comment (aleksandra-malinowska) [reply cluster-autoscaler/core/utils.go]: Done

**Pull request merged**